File: 0836-rectangle-overlap/0836-rectangle-overlap.cpp
class Solution {
public:
    bool isRectangleOverlap(vector<int>& rec1, vector<int>& rec2) {
        int x1 = rec1[0], y1 = rec1[1], x2 = rec1[2], y2 = rec1[3];
	    int x3 = rec2[0], y3 = rec2[1], x4 = rec2[2], y4 = rec2[3];
        return ( min(x2,x4) > max(x1,x3) && min(y2,y4) > max(y1,y3));
    }
};

File: 0836-rectangle-overlap/0836-rectangle-overlap.py
class Solution:
    def isRectangleOverlap(self, rec1: List[int], rec2: List[int]) -> bool:
        x1, y1, x2, y2 = rec1
        x3, y3, x4, y4 = rec2
        a = max(min(x2,x4) > max(x1,x3),0)
        b = max(min(y2,y4) > max(y1,y3), 0)
        return a * b > 0